File: mobile/ios/packages/town-crier-domain/Sources/ValueObjects/ZoneNotificationPreferences.swift
/// Per-zone notification preferences controlling which alert channels the user receives
/// for a specific watch zone.
///
/// The API shape matches `GET/PUT /v1/me/watch-zones/{zoneId}/preferences` (see
/// `GetZonePreferencesResult` / `UpdateZonePreferencesCommand` on the API). The four
/// per-channel toggles default to true so newly-created zones opt in to all alerts;
/// free-tier downgrades are applied at dispatch time on the server.
public struct ZoneNotificationPreferences: Equatable, Sendable {
  public let zoneId: String
  public let newApplicationPush: Bool
  public let newApplicationEmail: Bool
  public let decisionPush: Bool
  public let decisionEmail: Bool

  public init(
    zoneId: String,
    newApplicationPush: Bool = true,
    newApplicationEmail: Bool = true,
    decisionPush: Bool = true,
    decisionEmail: Bool = true
  ) {
    self.zoneId = zoneId
    self.newApplicationPush = newApplicationPush
    self.newApplicationEmail = newApplicationEmail
    self.decisionPush = decisionPush
    self.decisionEmail = decisionEmail
  }
}

File: mobile/android/domain/src/main/kotlin/uk/towncrierapp/domain/watchzones/ZoneNotificationPreferences.kt
package uk.towncrierapp.domain.watchzones

/**
 * Per-zone notification preferences controlling which alert channels the
 * user receives for a specific watch zone. Wire shape matches
 * `GET/PUT /v1/me/watch-zones/{zoneId}/preferences`. All four per-channel
 * toggles default to `true` so newly-created zones opt in to every alert;
 * free-tier downgrades are applied at dispatch time on the server. Port of
 * iOS `ZoneNotificationPreferences`.
 */
public data class ZoneNotificationPreferences(
    public val zoneId: WatchZoneId,
    public val newApplicationPush: Boolean = true,
    public val newApplicationEmail: Boolean = true,
    public val decisionPush: Boolean = true,
    public val decisionEmail: Boolean = true,
)
